### ISSUE TITLE
Freedom doesn't work in XaAES

### ISSUE THREAD

**Issue by mikrosk** [freemint/freemint#380]
Let's make @swe-yc happy and put it in our "backlog":

_Freedom is just one example. I understand not much ppl use freedom and maybe little people know why i like it: apart from looks, and many options, including skiping folders, etc, i like that its acc, and in TOS  one can open 2 or more instances of freedom, and use it as file explorer, copy app, etc. Very handy inside apps in TOS, etc._ 

Yes, we do have the note in the wiki / readme: https://github.com/freemint/freemint/blob/e0133e6f98b00e02dcb1dcb5cf2eaea4d44bb611/doc/ChangeLog.old/readme.1.16.0#L359 but that's not really a solution, is it.

If I understood Swe right, Freedom didn't work at all but now it does to some extent.

It works with N.AES and most likely in MyAES, too (@Landemarre feel free to correct me).

**Comment by skaftetryne** [comment]
Which version of Freedom? I have used Freedom with XaAES previously, but only 1.15. Freedom 2.x doesn't work reliably on any OS IMO. I'll try to remember to test this with the current kernel.

**Comment by th-otto** [comment]
Freedom (even if it works with TOS) is an ugly hack. It does not make sense to support it. What does it offer that builtin fileselector of XaAES cannot do?

**Comment by mikrosk** [comment]
> What does it offer that the builtin fileselector of XaAES cannot do?

See above. It really is a powerful tool, I also used to use it in TOS and MagiC. IIRC it has CPX manager, full featured file manager, ran in a separate non-modal window etc etc. So I don't blame anyone who wants it back.

**Comment by mikrosk** [comment]
> If I understood Swe right, Freedom didn't work at all but now it does to some extent.

I have more information about this:

> Which version of Freedom? I have used Freedom with XaAES previously, but only 1.15.

_Yes, and it works. There is some complainig from XaAES, but it works. 1.15 freedom works in MultiTOS, but fails in unix... F2 is crap_

So it would seem that there really isn't that much work left? Maybe just check the warnings, similarly as @th-otto did in https://github.com/freemint/freemint/commit/8ee8594b8ea00615ce98706aaa480f5880671849.

**Comment by skaftetryne** [comment]
> It really is a powerful tool, I also used to use it in TOS and MagiC. IIRC it has CPX manager, full featured file manager, ran in a separate non-modal window etc etc. So I don't blame anyone who wants it back.

This is Freedom 2. I remember being so disappointed with it after having used 1.15 for a long time. Freedom 2 tried to do a lot of stuff that's totally unrelated to the actual task of a file selector, and it also did it in a very unstable way. I don't see any reason for using such a tool in a multitasking system, as there are dedicated applications that does both file management and control panel functions a lot better.

**Comment by mikrosk** [comment]
Fair enough. I think I started with 1.15 as well but then switched to 2.0 but since I used it mostly in TOS, I didn't care that much about its compatibility.

Still, I think it's worth some effort to make 1.15 working on XaAES. MyAES doesn't support it at all, for instance.

**Comment by Landemarre** [comment]
> Let's make @swe-yc happy and put it in our "backlog":
> 
> _Freedom is just one example. I understand not much ppl use freedom and maybe little people know why i like it: apart from looks, and many options, including skiping folders, etc, i like that its acc, and in TOS one can open 2 or more instances of freedom, and use it as file explorer, copy app, etc. Very handy inside apps in TOS, etc._
> 
> Yes, we do have the note in the wiki / readme:
> 
> https://github.com/freemint/freemint/blob/e0133e6f98b00e02dcb1dcb5cf2eaea4d44bb611/doc/ChangeLog.old/readme.1.16.0#L359
> but that's not really a solution, is it.
> 
> If I understood Swe right, Freedom didn't work at all but now it does to some extent.
> 
> It works with N.AES and most likely in MyAES, too (@Landemarre feel free to correct me).

Don't know never test Freedom under mint 1.15 was test only under 1.19 and always crash, I just  try on my old V4 multisys config if freedom work with MyAES but if it not crash, it not work too, CPX said me not found config but I don't know where put file, never use this software. Will try again tomorrow.

**Comment by Landemarre** [comment]
Just do test under Mint 1.15.12 and V4SA, Freedom 1.15 and 2.05 work under NAES and AES 4.1, they not work under MyAES (do nothing) even if in memory, call of freedom.acc 2.05 open a window but nothing display on it, close it and move it work, very strange, this window not appear when application call fileselect, the usual fileselector is open.
So not work at all under MyAES.

**Comment by skaftetryne** [comment]
I just tried Freedom 1.15 with a very current kernel/XaAES on my Falcon060/SV. Memory protection is enabled.I have not really used it since my N.AES-days, but I still have my old Freedom setup on my bootdrive. I started ffsel.prg first and then freedom.app, both from the desktop. Some rudimentary testing indicates that all features are working. Lots of features which makes sense in a single-tasking OS, but not so much in a multitasking OS where you have the desktop available at all times.

**Comment by mikrosk** [comment]
Thanks for testing @skaftetryne. I guess the only thing left to test is that weird `INIT=` scenario whether it is really true or some hiccup in @swe-yc's setup.

**Comment by skaftetryne** [comment]
I'm afraid I don't have a suitable setup for that.

**Comment by swe-yc** [comment]
#response_container_BBPPID{font-family: initial; font-size:initial; color: initial;} Yes, Freedom 2 is a big mess.  In any case, starting ffsel.prg from aes config file (xaaes and myaes) solve the problem in "multitos" mode (aka GEM=), but if AES is started from full unix mint (INIT=) then it wont work.  I rechecked stuff, and i cant find anything wrong...                                                                                                                                                                          From: ***@***.***: 18 September 2024 11:57To: ***@***.*** to: ***@***.***: ***@***.***; ***@***.***: Re: [freemint/freemint] Freedom 1.15 nearly works flawlessly in XaAES (Issue #380)  

It really is a powerful tool, I also used to use it in TOS and MagiC. IIRC it has CPX manager, full featured file manager, ran in a separate non-modal window etc etc. So I don't blame anyone who wants it back.

This is Freedom 2. I remember being so disappointed with it after having used 1.15 for a long time. Freedom 2 tried to do a lot of stuff that's totally unrelated to the actual task of a file selector, and it also did it in a very unstable way. I don't see any reason for using such a tool in a multitasking system, as there are dedicated applications that does both file management and control panel functions a lot better.

**Comment by mikrosk** [comment]
@Landemarre perhaps your Freedom setup is somewhat different (as you couldn't get it working), you see, Swe's MyAES setup works nicely with Freedom.

**Comment by swe-yc** [comment]
#response_container_BBPPID{font-family: initial;font-size: initial;color: initial;} Keyword here is to start ffsel.prg after AES, so, manually or from myaes.cnf.... That should make it work.  About Freedom: some features arent crucial in mtasking, as skaftetryne already said, but its still very powerfull. Click on anything, mid path, sliders, etc, then right click anything, etc, etc...                                                                                                                                                                          From: ***@***.***: 20 September 2024 09:10To: ***@***.*** to: ***@***.***: ***@***.***; ***@***.***: Re: [freemint/freemint] Freedom 1.15 nearly works flawlessly in XaAES (Issue #380)  
@Landemarre perhaps your Freedom setup is somewhat different (as you couldn't get it working), you see, Swe's MyAES setup works nicely with Freedom.

**Comment by mikrosk** [comment]
After a bit of digging I found: https://mikro.naprvyraz.sk/mint/199909/msg00053.html and https://mikro.naprvyraz.sk/mint/199909/msg00301.html. This could explain the weirdness observed by @swe-yc, too.

Closing, nothing to do here. Freedom really isn't proper software to run in multitasking. Sorry Swe, @skaftetryne was equally sad 27 years ago. ;)
